plugins/edge-config/index.ts

```diff
@@ -10,38 +10,82 @@
   governing permissions and limitations under the License.
 */
 
-import { Configuration } from '@adobe/griffon-toolkit-aep-mobile';
+import { SharedStateConfig } from '@adobe/griffon-toolkit-aep-mobile';
 import { Event } from '@adobe/griffon-toolkit-common';
 import { ValidationPluginResult } from '../../types/validationPlugin';
 
 (function (events: Event[]): ValidationPluginResult {
   const { toolkit: kit } = window.griffon;
-  const { configuration } = kit['aep-mobile'];
+  const { sharedStateConfig } = kit['aep-mobile'];
   const configEvents = kit.match(
-    configuration.matcher,
+    sharedStateConfig.matcher,
     events
-  ) as Configuration[];
+  ) as SharedStateConfig[];
 
-  const hasAllConfig = (event: Configuration) =>
-    configuration.getEventDataKey('"edge.configId"', event);
+  const hasEdgeConfigId = window.griffon.toolkit.search('payload.metadata."state.data"."edge.configId"' , configEvents[0]);
+
+  const hasEdgeConfigDomain = window.griffon.toolkit.search('payload.metadata."state.data"."edge.domain"' , configEvents[0]);
+
+  const regex = /^(?:[a-z0-9](?:[a-z0-9-]{0,61}[a-z0-9])?\.)+[a-z0-9][a-z0-9-]{0,61}[a-z0-9]$/gi;
+
+  var validEdgeConfigDomain = false;
+
+  //If edge domain is existed, check if it is in the right format
+  if (hasEdgeConfigDomain) {
+    if (regex.test(hasEdgeConfigDomain) && !hasEdgeConfigDomain.includes("http" || "https")) {
+    validEdgeConfigDomain = true;
+    } else {
+      validEdgeConfigDomain = false;
+    }
+  }
```

Comment:
can be simplified to one line and you should be able to drop the check for http/https as your regex takes care of that:
```
//If edge domain is existed, check if it is in the right format
var validEdgeConfigDomain = hasEdgeConfigDomain && regex.test(hasEdgeConfigDomain)
```

Reply:
I like this. use `const` with this approach